layouts/default.vue

```diff
@@ -1,3 +1,34 @@
+<script setup lang="ts">
+type Theme = 'light' | 'dark'
+
+interface Message {
+  type: 'theme'
+  theme: Theme
+}
+
+function setTheme(theme: Theme) {
+  document.documentElement.classList.remove('light', 'dark')
+  document.documentElement.classList.add(theme)
+}
+
+const { initialTheme } = useRoute().query
+
+onMounted(() => {
+  if (initialTheme && (initialTheme === 'light' || initialTheme === 'dark')) {
+    setTheme(initialTheme)
+  }
+
+  window.addEventListener('message', (event: MessageEvent<Message>) => {
```

Comment:
We forgot to remove the listener on unmount. Best to use [`useEventListener`](https://vueuse.org/core/useEventListener/#useeventlistener) from VueUse (again, always check if they have helpers you need, they usually do).

I understand that coming from React, you're used to having to do everything by hand :).

Reply:
I was gonna look up how to do cleanup with `onMounted` but forgot!

Awesome about `VueUse`, I'll try to (vue)use it more often!